app/models/host.rb

```diff
@@ -1279,6 +1279,45 @@ def refresh_firewall_rules
   def refresh_advanced_settings
   end
 
+  def refresh_network_interfaces(ssu)
+    smartstate_network_ports = MiqLinux::Utils.parse_network_interface_list(ssu.shell_exec("ip a"))
+
+    neutron_network_ports = network_ports.where(:source => :neutron).each_with_object({}) do |network_port, obj|
+      obj[network_port.mac_address] = network_port
+    end
+    neutron_cloud_subnets = ext_management_system.cloud_subnets
+    hashes = []
+
+    smartstate_network_ports.each do |network_port|
+      existing_network_port = neutron_network_ports[network_port[:mac_address]]
+      if existing_network_port.blank?
+        cloud_subnet = neutron_cloud_subnets.detect do |neutron_cloud_subnet|
+          if neutron_cloud_subnet.ip_version == 4
+            IPAddr.new(neutron_cloud_subnet.cidr).include?(network_port[:fixed_ip])
+          else
+            IPAddr.new(neutron_cloud_subnet.cidr).include?(network_port[:fixed_ipv6])
+          end
+        end
+
+        hashes << {:name          => network_port[:name],
+                   :mac_address   => network_port[:mac_address],
+                   :cloud_subnet  => cloud_subnet,
+                   :cloud_network => cloud_subnet.try(:cloud_network),
+                   :device        => self,
+                   :fixed_ips     => {:subnet_id     => nil,
+                                      :ip_address    => network_port[:fixed_ip],
+                                      :ip_address_v6 => network_port[:fixed_ipv6]}}
+      else
+        # Just updating a names of network_ports refreshed fro Neutron, rest of attributes
+        # is handled in refresh section.
+        existing_network_port.update_attributes(:name => network_port[:name])
+      end
+    end
+    EmsRefresh.save_network_ports_inventory(ext_management_system, hashes, ext_management_system, :smartstate) if hashes
```

Comment:
`hashes` will never be `nil`, so checking `if hashes` at the end of the line here will never fail.

Maybe `unless hashes.blank?`